File: Ejercicios POO/PHP/Herencia y Composicion/Edificios.php
<?php

// Written By Ismael Heredia in the year 2016

class Edificio
{
    
    public $medidas;
    public $ambientes;
    
    public function __construct()
    {
        $this->medidas   = "";
        $this->ambientes = "";
    }
    
    public static function CreateEdificio($medidas,$ambientes)
    {
        $instance                 = new self();
        $instance->medidas   = $medidas;
        $instance->ambientes = $ambientes;
        return $instance;
    }
    
    public function getMedidas()
    {
        return $this->medidas;
    }
    
    public function setMedidas($medidas)
    {
        $this->medidas = $medidas;
    }
    
    public function getAmbientes()
    {
        return $this->ambientes;
    }
    
    public function setAmbientes($ambientes)
    {
        $this->ambientes = $ambientes;
    }
        
    public function __destruct()
    {
        $this->medidas   = "";
        $this->ambientes = "";
    }
    
    public function DatosEdificio()
    {
    	return "-- Datos del terreno --" . "\n\n"
            . "[+] Medidas : " . $this->medidas . "\n"
            . "[+] Ambientes : " . $this->ambientes;
    }
    
}

?>

File: Ejercicios POO/Python/Herencia y Composicion/Edificios.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
# Written By Ismael Heredia in the year 2016

class Edificio(object):

   def __init__(self, medidas="", ambientes=""):
       self.__medidas = medidas
       self.__ambientes = ambientes
              
   def getMedidas(self):
       return self.__medidas

   def setMedidas(self, medidas = ""):
       self.__medidas = medidas

   def getAmbientes(self):
       return self.__ambientes

   def setAmbientes(self, ambientes = ""):
       self.__ambientes = ambientes
           
   def DatosEdificio(self):
       return "-- Datos del edificio --\n\n[+] Medidas : %s\n[+] Ambientes : %d" % (self.__medidas,self.__ambientes)
          
   def destroy(self):
       self.__medidas = None
       self.__ambientes = None

